src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -1051,7 +1051,7 @@ void ClusterTaskManager::TryLocalInfeasibleTaskScheduling() {
         placement_resources,
         /*requires_object_store_memory=*/false,
         task.GetTaskSpecification().IsActorCreationTask(),
-        /*force_spillback=*/false, &_unused, &is_infeasible);
+        /*force_spillback=*/false, work->grant_or_reject, &_unused, &is_infeasible);
```

Comment:
I think it might be simpler if we don't pass the grant_or_reject flag to the ClusterResourceScheduler and just decide which ClusterResourceScheduler methods to call from this class. That way:
1. This class can control when we actually need to call the full policy.
2. We have to check the flag in this class anyway, so this way we keep the flag contained to this class.

Reply:
Actually after a second thought, I think it's probably still better to have it inside ClusterResourceScheduler since it's about scheduling and ClusterResourceScheduler should be responsible for choosing what scheduling policy to use based on the input scheduling constraints (e.g force_spillback, grant_or_reject) and expose a single GetBestSchedulableNode API. What do you think?

Reply:
Hmm I still think it's nicer to have it in the ClusterTaskManager (which is also about scheduling, maybe this is part of the issue), so that we can keep the logic more modular. So for example, now we have to test both ClusterResourceScheduler and ClusterTaskManager logic to make sure grant_or_reject is respected, whereas if we kept it only in ClusterTaskManager, we would only have to test that class.

But I think it's pretty minor, so I'll leave it up to you.

Reply:
Ok, Thinking ClusterTaskManager is also about scheduling makes more sense to me now. Will change it.